CMake/hoomd/HOOMDCUDASetup.cmake

```diff
@@ -1,73 +1,39 @@
-# Maintainer: joaander
+option(ENABLE_CUDA "Enable the compilation of the CUDA GPU code" off)
+option(ENABLE_NVTOOLS "Enable NVTools profiler integration" off)
 
-##################################
-## Find CUDA
-# If CUDA is enabled, set it up
-if (ENABLE_CUDA)
-    find_package(CUDA 8.0 REQUIRED)
-    find_package(Thrust 1.5.0 REQUIRED)
-
-    # first thrust, then CUDA (to allow for local thrust installation
-    # that overrides CUDA toolkit)
-    include_directories(${THRUST_INCLUDE_DIR})
-    include_directories(${CUDA_INCLUDE_DIRS})
-
-    get_directory_property(DIRS INCLUDE_DIRECTORIES SYSTEM)
-    # hide some variables users don't need to see
-    mark_as_advanced(CUDA_SDK_ROOT_DIR)
-    if (CUDA_TOOLKIT_ROOT_DIR)
-        mark_as_advanced(CUDA_TOOLKIT_ROOT_DIR)
-    endif (CUDA_TOOLKIT_ROOT_DIR)
-    mark_as_advanced(CUDA_VERBOSE_BUILD)
-    mark_as_advanced(CUDA_BUILD_EMULATION)
-    mark_as_advanced(CUDA_HOST_COMPILER)
-    mark_as_advanced(CUDA_dl_LIBRARY)
-    mark_as_advanced(CUDA_rt_LIBRARY)
-    mark_as_advanced(THRUST_INCLUDE_DIR)
-
-    # make sure the cudadevrt library has been found (needed in old FindCUDA)
-    if (NOT CUDA_cudadevrt_LIBRARY)
-        get_filename_component(CUDA_LIB_PATH ${CUDA_CUDART_LIBRARY} DIRECTORY)
-        find_library(CUDA_cudadevrt_LIBRARY cudadevrt PATHS ${CUDA_LIB_PATH} NO_DEFAULT_PATH)
-        mark_as_advanced(CUDA_cudadevrt_LIBRARY)
-    endif (NOT CUDA_cudadevrt_LIBRARY)
+option(ALWAYS_USE_MANAGED_MEMORY "Use CUDA managed memory also when running on single GPU" OFF)
+MARK_AS_ADVANCED(ALWAYS_USE_MANAGED_MEMORY)
 
-    if (ENABLE_NVTOOLS)
-        find_library(CUDA_nvToolsExt_LIBRARY
-                     NAMES nvToolsExt
-                     PATHS "${CUDA_TOOLKIT_ROOT_DIR}/lib64"
-                           "${CUDA_TOOLKIT_ROOT_DIR}/lib"
-                     ENV CUDA_LIB_PATH
-                     DOC "nvTools library"
-                     NO_DEFAULT_PATH
-                     )
-
-        mark_as_advanced(CUDA_nvToolsExt_LIBRARY)
+if (ENABLE_CUDA)
+    enable_language(CUDA)
+    if (CMAKE_CUDA_COMPILER_VERSION VERSION_LESS 9.0)
+        message(SEND_ERROR "HOOMD-blue requires CUDA 9.0 or newer")
     endif()
+
+    find_package(CUDALibs REQUIRED)
 endif (ENABLE_CUDA)
 
 # setup CUDA compile options
 if (ENABLE_CUDA)
     # supress warnings in random123
-    list(APPEND CUDA_NVCC_FLAGS "-Xcudafe;--diag_suppress=code_is_unreachable")
+    set(CMAKE_CUDA_FLAGS "${CMAKE_CUDA_FLAGS} -Xcudafe --diag_suppress=code_is_unreachable")
 
     # setup nvcc to build for all CUDA architectures. Allow user to modify the list if desired
-    if (CUDA_VERSION VERSION_GREATER 8.99)
+    if (CMAKE_CUDA_COMPILER_VERSION VERSION_GREATER 8.99)
```

Comment:
Indeed. I'm leaving this for now so the if/then/else structure is there to modify when future architectures are added (e.g. in cuda 11).

I removed CUDA 8 support because the CUDA 8 docker image is based on ubuntu16.04 which does not have a new enough CMake to build this branch. We can CUDA 8 support back in if needed - though by the time hoomd v3 is released it will be quite old and likely not worth the effort to maintain.